sklearn/tests/test_cross_validation.py

```diff
@@ -483,6 +483,36 @@ def test_predefinedsplit_with_kfold_split():
     assert_array_equal(ps_test, kf_test)
 
 
+def test_shuffle_labels_out():
+    ys = [np.array([1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 3, 3]),
+          np.array([0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3]),
+          np.array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2]),
+          np.array([1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4]),
+          ]
+
+    for y in ys:
+        n_iter = 6
+        slo = cval.ShuffleLabelsOut(y, n_iter, test_size=0.33,
```

Comment:
> train size is not tested

I don't think we need to add an explicit test for this, but I'm happy to do so if it fits a particular testing convention.

The train size is not directly tested, but the subsequent tests would fail since it checks for both empty intersection and that the sum of train and test equals the original length, so the net effect is the same.